src/main/java/org/opentripplanner/middleware/models/MonitoredTrip.java

```diff
@@ -235,13 +233,10 @@ public void initializeFromItineraryAndQueryParams() throws IllegalArgumentExcept
         int lastLegIndex = itinerary.legs.size() - 1;
         from = itinerary.legs.get(0).from;
         to = itinerary.legs.get(lastLegIndex).to;
-        Map<String, String> parsedQueryParams = this.parseQueryParams();
+        Map<String, String> parsedQueryParams = parseQueryParams();
 
-        // Update modes in query params with set derived from itinerary. This ensures that a potentially inaccurate value
-        // in query params is not used when monitoring trips in the future.
-        Set<String> modes = ItineraryUtils.deriveModesFromItinerary(itinerary);
-        parsedQueryParams.put(MODE_PARAM, String.join(",", modes));
-        this.queryParams = ItineraryUtils.toQueryString(parsedQueryParams);
+        // Update actual query modes from the itinerary, so that OTP requests query the correct modes desired by user.
+        updateModeInQueryParams(parsedQueryParams);
```

Comment:
My thought with `clearRealtimeInfo` was that it could be a generalized method that may eventually contain additional logic to clear out (or check) realtime info in the itinerary. `updateModeInQueryParams`, on the other hand, is a much more specialized method name that doesn't really leave room for it to grow/change.

I would be in favor of inlining the `itinerary.clearAlerts()` call rather than obscuring these 2-3 lines of code in `updateModeInQueryParams`.

Reply:
@evansiroky we need a tie breaker!

Reply:
Since `updateModeInQueryParams` seems to only be called from here and it's a small method, I think inlining it makes sense.

Reply:
Okay. Done in https://github.com/ibi-group/otp-middleware/pull/104/commits/47bb6c96c8650583e7496115f1f1fb8c18e8176b. Thanks for the input.